parsers/thueringen.py

```diff
@@ -0,0 +1,169 @@
+
+	sel_week = document.find('select', {"name": 'selWeek'})
+
+	options = sel_week.find_all('option')
+
+	for option in options:
+		week = option['value']
+		if week != '0':
+			yield week
+
+
+def parse_fees(document):
+	fees_regex = re.compile('Bedienstete.*')
+
+	fees = document.find_all('p', class_='MsoNormal', string=fees_regex)
+
+	for fee in fees:
+
+		fee_strings = fees_regex.findall(fee.text)
+		for amount_candidate in fee_strings:
+			amount_strings = amount_regex.findall(amount_candidate)
+			if len(amount_strings) != 2:
+				continue
+
+			employees_fee = float(amount_strings[0].replace(',', '.'))
+			guests_fee = float(amount_strings[1].replace(',', '.'))
+			return employees_fee, guests_fee
+
+	return None, None
+
+
+def parse_ingredients(document):
+	groups_regex = re.compile('([\w\d*]+):\s+(.*)')
+	groups = dict()
+
+	for table in document.select('div.kontextbox > table:nth-of-type(1)'):
+		for s in table.stripped_strings:
+			if groups_regex.search(s):
+				split = s.split(':')
+				groups[split[0]] = split[1]
+
+	return groups
+
+
+def parse_meals(day, groups):
+	ingredients_regex = re.compile('Inhalt:.*')
+
+	meals_t_rows = day.find_all('tr')
+
+	for meal_data in meals_t_rows:
+		meal_t_datas = meal_data.find_all('td')
+		if len(meal_t_datas) != 3:
+			continue
+
+		category = meal_t_datas[0].text
+		notes = []
+		ingredients = ingredients_regex.findall(meal_t_datas[1].text)
+
+		if len(ingredients) > 0:
+			ingredients_list = ingredients[0].strip('Inhalt: ').strip().split(',')
+
+			notes = [groups[note].strip() for note in ingredients_list if note in groups]
+
+		main_dish = ingredients_regex.sub('', meal_t_datas[1].text).strip()
+
+		students_fee_string = amount_regex.findall(meal_t_datas[2].text)
+		if len(students_fee_string) != 1:
+			continue
+
+		students_fee = float(students_fee_string[0].replace(',', '.'))
+		costs = {'student': students_fee}
+
+		yield (main_dish, notes, costs, category)
+
+
+def parse_url(url, today=False):
+
+	days_regex = re.compile('day_\d$')
+
+	canteen = LazyBuilder()
+
+	content = urlopen(url).read()
+	document = parse(content, 'lxml')
+
+	available_weeks = parse_available_weeks(document)
+	employees_fee, guests_fee = parse_fees(document)
+	groups = parse_ingredients(document)
+
+	for week in available_weeks:
+
+		content = urlopen("{}?selWeek={}".format(url, week)).read()
+
+		document = parse(content, 'lxml')
+
+		mensa_start_date = parse_start_date(document)
+
+		day_divs = document.find_all('div', id=days_regex)
+
+		for day in day_divs:
+			day_id = int(day['id'][-1:])
+
+			current_date = mensa_start_date + timedelta(days=day_id-2)
+
+			meals = parse_meals(day, groups)
+			for meal in meals:
+				main_dish, notes, costs, category = meal
+
+				if employees_fee is not None:
+					costs['employee'] = costs['student'] + employees_fee
+				if guests_fee is not None:
+					costs['other'] = costs['student'] + guests_fee
+				if today and current_date != date.today():
```

Comment:
This still fetches all sites, even when just the first one is needed. You need another check that only fetches the current site for today feeds (see also the comment above).